corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -169,6 +169,7 @@ def get_form_name_from_last_submission_for_xmlns(domain, xmlns):
         .sort('received_on', desc=True)
         .source(['form.@name'])
         .size(1)
+        .non_null('form.@name')
```

Comment:
FWIW pre @snopoke's PR https://github.com/dimagi/commcare-hq/pull/26420 it just got the first one and returned `None` if the `form` property didn't have a `@name` property, whereas this now makes the extra effort to find a `@name` if any form has it. Not sure there's a difference in practice and/or if this is maybe better, just wanted to point out the subtle change in case it's relevant later

Reply:
I think this change is OK though it does push more work onto ES. If the most recently received for does not have a name then it's unlikely any others would. A non-conclusive test on ICDS showed a 3x query time increase for the additional filter when applied to system forms (subsequent queries were much faster due to caching but the cache didn't last for long).

I'm not going to motivate to change this now but may revisit at a later point. At ICDS scale it makes less and less sense to get this kind of info by querying an index with 1.2B forms in it.